### PR TITLE
Magiclysm: improve confirmation menu

### PR DESCRIPTION
**Pull request by KorGgenT**
#### Summary

```SUMMARY: Mods "Improve spell learning confirmation menu"```
#### Purpose of change

The confirmation menu wasn't clear before: i added in the list of spell classes you could no longer learn from.
image

### CODE REVIEW

File: src/magic.cpp [hunk 1]
```diff
@@ -785,9 +785,23 @@ void known_magic::learn_spell( const spell_type *sp, player &p, bool force )
...
+                    trait_cancel = string_format( "%s, %s", trait_cancel, cancel->name() );
+                }
+                if( cancel == sp->spell_class->cancels.back() ) {
+                    trait_cancel += ".";
+                }
+            }
             if( query_yn(
-                    _( "Learning this spell will make you a %s and lock you out of other unique spells.\nContinue?" ),
-                    sp->spell_class.obj().name() ) ) {
+                    _( "Learning this spell will make you a %s and lock you out of %s\nContinue?" ),
```
Comment (LaVeyanFiend) [inline src/magic.cpp]:
```suggestion
                    _( "Learning this spell will make you a %s and lock you out of %s.\nContinue?" ),
```

Comment (KorGgenT) [reply src/magic.cpp]: that period is already added. i noticed it in the screenshot and then fixed it, but didn't update the screenshot.

Comment (KorGgenT) [reply src/magic.cpp]: (on line 799 above)

Comment (AMurkin) [reply src/magic.cpp]: It would be better to place it here imo. For the translations and simpler code.

**Pull request merged**